app/provider/app.go

```diff
@@ -171,6 +171,14 @@ func (a *app) doCheckTx(ctx apptypes.Context, tx *types.TxCreateProvider) tmtype
 		return tmtypes.ResponseCheckTx{Code: code.INVALID_TRANSACTION, Log: "invalid nonce"}
 	}
 
+	// XXX: more address validation
```

Comment:
you can use [this](https://golang.org/pkg/net/url/#Parse) and make sure that there's a scheme & host.